### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by SaeqaSultani**
- Add the following validations:
  1-  For the User model:
        Name must not be blank.
       PostsCounter must be an integer greater than or equal to zero.
 2- For the Post model:
      Title must not be blank.
     Title must not exceed 250 characters.
    CommentsCounter must be an integer greater than or equal to zero.
    LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,224 @@
...
+
+>
+
+<details>
+  <summary>Framework</summary>
+  <ul>
+    <li></li>
+  </ul>
+</details>
+
```
Comment (BenMukebo) [inline README.md]: - [ ] Let's add Ruby on Rails as a Tech Stack Framework. ✍🏻 

image

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    expect(subject).to_not be_valid
+  end
+
+  it 'likes counter should be integer' do
+    subject.likes_counters = 'c'
+    expect(subject).to_not be_valid
+  end
+
+  it 'likes counter should be greater than or equal to zero' do
+    subject.likes_counters = -2
```
Comment (BenMukebo) [inline spec/models/post_spec.rb:44]: - [ ] According to the project requirement, 👇🏻 
1. You should write unit tests for all your models: `User`, `Post`, `Comment`, and `Like` models. Kindly add unit tests for the  `Comment`, and `Like` models.
2. Any custom methods in the models should have a corresponding unit test. You're missing adding unit tests of the  `post_counter and recent_comments` methods of your Post model, the ` comment_counter` method of your Comment model, and the `like_counter` method. Please address this ✍🏻 

im2

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @SaeqaSultani,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- User model has validations. ✔️ 
- Post model has validations.✔️ 
- Gitflow was correctly used.  ✔️   
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb
Comment (fobadara) [inline spec/models/post_spec.rb:1]: [REQUIRED] As stated in the project requirements and by the previous reviewer you should write tests for custom methods . Kindly include tests for `recent_comments` and `post_counter`

File: spec/models/comment_spec.rb
Comment (fobadara) [inline spec/models/comment_spec.rb:1]: [REQUIRED] As stated in the project requirements and by the previous reviewer you should write tests for custom methods . Kindly include tests for  `comment_counter`

File: spec/models/like_spec.rb
Comment (fobadara) [inline spec/models/like_spec.rb:1]: [REQUIRED] As stated in the project requirements and by the previous reviewer you should write tests for custom methods . Kindly include tests for `like_counter`

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @SaeqaSultani, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet.


 ## Highlights
✅ Linters are passing

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  describe 'post_counter' do
+    it 'increments the author posts_counter by 0' do
+      user = User.create(name: 'User 1', posts_counter: 0)
+      Post.create(title: 'Post 1', author_id: user, comments_counter: 0, likes_counters: 0)
+      expect(user.posts_counter).to eq(0)
+    end
+  end
+
+  it 'Should return 0 posts' do
```
Comment (VICTOR-CHUKWU) [inline spec/models/post_spec.rb:56]: - [ ] [OPTIONAL] i guess this is test for recent comment method, the test description is confusing, kindly name it properly :+1:

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @SaeqaSultani , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 3 above)
Comment (SaeqaSultani) [reply spec/models/post_spec.rb:56]: ok, I will correct it.
thanks.

**Pull request merged**